flat_mutation_reader.hh

```diff
@@ -360,6 +360,19 @@ public:
         return _impl->consume(std::move(consumer), timeout);
     }
 
+    class filter {
```

Comment:
What is the advantage of this class? It looks like a mere wrapper over `std::function`.

Reply:
As @tgrabiec suggested, it eventually defines to predicates, one for `dht::decorated_key` and the other for `mutation_fragment`.  This way we can still pass one filter to `consume_in_thread` and friends rather than two.